tiledb/sm/query/readers/dense_reader.cc

```diff
@@ -310,6 +313,25 @@ Status DenseReader::dense_read() {
   if (condition_.has_value()) {
     qc_loaded_attr_names_set_ = condition_->field_names();
   }
+  qc_loaded_attr_names_.clear();
+  qc_loaded_attr_names_.reserve(qc_loaded_attr_names_set_.size());
+  for (auto& name : qc_loaded_attr_names_set_) {
+    qc_loaded_attr_names_.emplace_back(name);
+  }
+
+  // Load per fragment tile offsets memory usage.
+  per_frag_tile_offsets_usage_ = tile_offset_sizes();
+
+  // Compute total tile offsets sizes.
+  auto total_tile_offsets_sizes = std::accumulate(
+      per_frag_tile_offsets_usage_.begin(),
+      per_frag_tile_offsets_usage_.end(),
+      static_cast<uint64_t>(0));
+  if (total_tile_offsets_sizes >
+      memory_budget_ - array_memory_tracker_->get_memory_usage()) {
```

Comment:
Indeed, the `get_memory_usage()` function does not subtract the tile offsets from the new "counter" memory tracking, and the old tracking system doesn't account for them either. However, `array_memory_tracker_->get_memory_available()` does.

Reply:
This is desired. There are some memory structures allocated for tile offsets at this time and we don't want them to be included.

Reply:
I thought the only structure allocated for tile offsets is the ```counters```

Reply:
Other readers have tile offsets loaded at the time they are ready to process tiles.

Reply:
"I thought the only structure allocated for tile offsets memory is the counters"

Yes they are, but get_memory_available() will subtract that counter from the available memory. Since our calculation include everything, we should not remove the tile offsets counter from the available memory.

Reply:
Why we don't want to subtract it from available memory? Isn't that memory taken by some of our structures, so it cannot be used for loading offsets? We are we tracking those structures then?
If that's just some confusing design we wouldn't like to address as part of this ticket, let's make sure we address it in some follow up.

Reply:
We don't subtract it from the available memory because it's included in total_tile_offsets_sizes... We don't want to include it twice. There's already stories to reconcile everything and get rid of the legacy memory tracker which confuses everything. But before we can do that, we need to complete the partial tile offsets loading for all readers as well as partial rtree loading.